packages/patternfly-4/react-core/src/components/Card/CardHead.tsx

```diff
@@ -13,8 +19,37 @@ export const CardHead: React.FunctionComponent<CardHeadProps> = ({
   children = null,
   className = '',
   ...props
-}: CardHeadProps) => (
-  <div className={css(styles.cardHead, className)} {...props}>
-    {children}
-  </div>
-);
+}: CardHeadProps) => {
+
+  const actionChildren = React.Children.toArray(children).filter(child => isReactElement(child) && child.type === CardActions)
+  const mainChildren = React.Children.toArray(children).filter(child => !isReactElement(child) || child.type !== CardActions)
+  return (
+    <div className={css(styles.cardHead, className)} {...props}>
+      <CardHeadMain>
+        {mainChildren}
+      </CardHeadMain>
+      {actionChildren}
```

Comment:
There is an implied order here. What if I've defined `<CardActions>` first? 

I believe the output would be quite unexpected if users provided a layout like this:
```

<CardHead>
  <img .../>
  <CardActions>
  <img .../>
</CardHead>
```

@mcoker I don't know that we want to do this with React composition? I believe users should be explicit when wrapping components with `<CardAction/>` and `<CardHeadMain/>`

Reply:
That's what I thought but I wasn't sure. However, every items that's not `CardActions` will be wrapped individually. If that's OK, then I have something to work on. Otherwise, we need to think more.

Reply:
It seems problematic to me

Reply:
It seems problematic to me (e.g., if someone wants a custom layout)

Reply:
@tlabaj @christiemolloy @mcoker - your thoughts.
I am trying automatically to wrap items under `CardHead` into `CardHeadMain` as you thought but I need more information.

Thanks.

Reply:
I agree with Dan here about the implied order.  This is too restrictive and can break current implementations.   I think in this case, that it is better to update the docs and guide the consumer to use the wrapper rather than restricting the order of the children.